src/test/groovy/org/gradle/android/ConfigurationCachingTest.groovy

```diff
@@ -0,0 +1,21 @@
+package org.gradle.android
+
+class ConfigurationCachingTest extends AbstractTest {
+    def "Can run with configuration cache"() {
+        given:
+        SimpleAndroidApp.builder(temporaryFolder.root, cacheDir)
+                .withAndroidVersion(Versions.latestAndroidVersion())
```

Comment:
I wonder if we should iterate over the "config-cache-compatible" versions of AGP.  At least initially this would only be the latest 4.2 version, but as new versions are released, we might want to ensure that we aren't somehow breaking an older version.  Dunno - might be over-paranoid testing.  Your call.